lightning/src/ln/channel.rs

```diff
@@ -4812,11 +4822,21 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<&'a K> for Channel<Signer>
 		let counterparty_dust_limit_satoshis = Readable::read(reader)?;
 		let holder_dust_limit_satoshis = Readable::read(reader)?;
 		let counterparty_max_htlc_value_in_flight_msat = Readable::read(reader)?;
-		let counterparty_selected_channel_reserve_satoshis = Readable::read(reader)?;
+		let mut counterparty_selected_channel_reserve_satoshis = Some(Readable::read(reader)?);
+		if counterparty_selected_channel_reserve_satoshis == Some(0) {
+			// Versions up to 0.0.98 had counterparty_selected_channel_reserve_satoshis as a
+			// non-option, writing 0 for what we now consider None.
+			counterparty_selected_channel_reserve_satoshis = None;
+		}
 		let counterparty_htlc_minimum_msat = Readable::read(reader)?;
 		let holder_htlc_minimum_msat = Readable::read(reader)?;
 		let counterparty_max_accepted_htlcs = Readable::read(reader)?;
-		let minimum_depth = Readable::read(reader)?;
+		let mut minimum_depth = Some(Readable::read(reader)?);
+		if minimum_depth == Some(0) {
+			// Versions up to 0.0.98 had minimum_depth as a non-option, writing 0 for what we now
+			// consider None.
+			minimum_depth = None;
+		}
```

Comment:
For 0-confirmation channels, will `0` be a legitimate depth?

Reply:
Unclear? Per Rusty's spec I assume not (it just changes the state machine...), but we should probably support it, yea. Note that if the second serialized value is `Some(0)` then we will store `Some(0)` its only when the first value is `0` that we will consider it `None`.